benchmarks/distributed/rpc/ps/metrics/MetricsLogger.py

```diff
@@ -0,0 +1,92 @@
+from .CPUMetric import CPUMetric
+from .CUDAMetric import CUDAMetric
+
+
+class MetricsLogger:
+
+    CUDA = "cuda"
+    CPU = "cpu"
+
+    def __init__(self, rank, metric_class, overwrite_metrics=False):
+        self.rank = rank
+        self.metric_class = metric_class.lower()
```

Comment:
this means we don't expect to use the same logger to log both CPU and CUDA metrics?

Reply:
I was thinking about this. CPUMetrics are not used in the experiments that I am implementing, but I can update the logger to specify logging CUDA or CPU.

```python

    def cpu_record_start(self, type, key, name):
        ...

    def cuda_record_start(self, type, key, name):
        ...            

```

I'll update the existing wrapper in the base classes and add a CPU wrapper.

Reply:
I ended up adding a CUDA parameter to indicate if the metric should be CUDA or CPU.
```python
    def record_start(self, type, key, name, cuda=True):
        self.__metrics_logger.record_start(
            type,
            key,
            name,
            cuda
        )
 ```